### ISSUE TITLE
error: "type" has no attribute "var"

### ISSUE THREAD

**Issue by pwilczynskiclearcode** [python/mypy#2694]
test.py:
```python
class A(object):
    var = 2

class B(A):
    var = 5


for cls in [A, B]:
    print(cls.var)
```

```
$ mypy test.py
test.py:9: error: "type" has no attribute "var"
```

```
$ python --version
Python 3.5.2
```

```
$ mypy --version
mypy 0.470
```

test.py:
```python
class A(object):
    var = 2  # type: int

class B(A):
    var = 5  # type: int


for cls in [A, B]:
    print(cls.var)
```

```
test.py:9: error: "type" has no attribute "var"
```

**Comment by gvanrossum** [comment]
There's a small bug in mypy's internal join() function that causes this. It doesn't realize that the "best" type for `[A, B]` is `List[Type[A]]`, so instead it comes up with `List[Type[Any]]` a.k.a. `List[type]`.

**Comment by gvanrossum** [comment]
A work-around would be:
```python
class A: ...
class B(A): ...
classes = [A, B]  # type: List[Type[A]]
for cls in classes:
    print(cls.var)
```

**Comment by pwilczynskiclearcode** [comment]
Guido, thank you. It helps to understand how does `mypy` works it this case. If only `mypy` would be more verbose when having problems with inferring types it would be easier to help it with explicit type declarations. Maybe introducing `-vv` showing `debug` or `trace` logs would help `mypy` users to deal with not yet covered corner cases.

**Comment by gvanrossum** [comment]
You're welcome. Such an option would indeed be useful, but it would be complicated to implement -- mypy often doesn't remember how it reached a certain conclusion. A useful debugging tool however is reveal_type() -- see the mypy docs for that.

**Comment by miedzinski** [comment]
This was fixed in 00359ade.

**Comment by ethanhs** [comment]
Confirmed as fixed.
